lib/Doctrine/ORM/Internal/Hydration/ObjectHydrator.php

```diff
@@ -401,7 +400,7 @@ protected function hydrateRowData(array $row, array &$result)
                     $reflFieldValue = $association->getValue($parentObject);
 
                     if (! $reflFieldValue || isset($this->hints[Query::HINT_REFRESH]) ||
-                        ($reflFieldValue instanceof Proxy && ! $reflFieldValue->__isInitialized())) {
+                        ($reflFieldValue instanceof GhostObjectInterface && ! $reflFieldValue->isProxyInitialized())) {
```

Comment:
It doesn't really matter, but this check is everywhere in the code. I would personally extract it to a helper function. Although an extra method call could have impact on the execution time, depending on the location of these checks.

Reply:
Sadly, this is a very common problems with the ORM: we cannot extract logic to methods without massive performance impact.

It is indeed very true that it is horrible, but php-src does not yet optimise cross-file calls.